nixos/modules/services/networking/syncthing.nix

```diff
@@ -10,6 +10,28 @@ let
   settingsFormat = pkgs.formats.json { };
   cleanedConfig = converge (filterAttrsRecursive (_: v: v != null && v != {})) cfg.settings;
 
+  # syncthing supports serving the gui over unix sockets. if that happens, the
+  # api is serverd over the unix socket as well. this module uses the api to
+  # merge the configuration. curl is used for this and supports unix sockets.
+  # however the arguments are slightly different from an http request. the
+  # function `curlAddressArgs` returns a given path the correct string to make
+  # curl connect over either http(s) or unix. string which tells curl how to
+  # connect to the address. this works for https and unix.
+  curlAddressArgs = path: let
+    matchResult = builtins.match "^unix://(.*)" cfg.guiAddress;
+  in
+    if matchResult == null
+    # for http and https we do nothing
+    then "${cfg.guiAddress}/${path}"
```

Comment:
We could be a little more explicit add do:
```
    then "http://${cfg.guiAddress}/${path}"
```

Reply:
I don't think that is a good idea, because synchting has the functionality to do https too. the http/https protocol should be part of cfg.guiAddress already.

Reply:
I might write this too but my reasoning here was that the Syncthing docs say that all HTTP requests will be redirected to HTTPS when `gui.tls = true`, so I don't think the user ever really needs to worry about adding a protocol prefix to the `guiAddress`.

https://docs.syncthing.net/users/config.html#config-option-gui.tls